debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -257,7 +257,16 @@ private Long getEndOffsetOfDbHistoryTopic(Long previousEndOffset, KafkaConsumer<
 
         return endOffset;
     }
-
+    
+    @Override
+    public boolean exists() {
+        boolean exists = false;
+        try (KafkaConsumer<String, String> historyConsumer = new KafkaConsumer<>(consumerConfig.asProperties());) {
+            exists = historyConsumer.listTopics().keySet().contains(topicName);
+        }
+        return exists;
+    }
+    
```

Comment:
There are probably many ways to implement exists(), I went with just checking the name exists in our list of topics. Not sure if this is the best way.

Reply:
What is the typical way you lose the history? Is it when the topic is deleted? If not is there another way how to check it, like is topic empty, can we build schema from content topic, etc.

Reply:
Yes, for Kafka this would be when the topic is deleted and in the case of FileDatabaseHistory if the file is deleted etc. I'm open to suggestion if there is a better way.

Reply:
We need a find another way - like a topic is empty. Remember - database history topic needs a special configuration
* Infinite (or very long) retention (no compaction!)
* Replication factor at least 3 for production
* Single partition
so it is necessary that it is created manually, not automatically. This means we cannot rely on missing a topic as we request the admin to create it!

Reply:
I am not sure this is safe - if the topic content is over retention period then it can be empty and at the same time offset will be non-zero. IMHO if possible we should get the first and last offset and compare them.

Reply:
@jpechane I can certainly check if endOffset - beginOffset == 0. But my understanding was this topic *requires* infinite retention so the beginOffset will always be 0 anyway. Let me know.